lib/smart_answer_flows/energy-grants-calculator.rb

```diff
@@ -470,6 +470,9 @@ def define
         precalculate :flat_type do
           flat_type
         end
+        precalculate :under_green_deal do
```

Comment:
@floehopper @erikse @ikennaokpala Happy to give a try to the calculator/policy object. It feels like a better approach. Not sure, though, if including it in this PR as we would have two different styles for precalculated values?